Zend/zend_object_handlers.c

```diff
@@ -326,13 +326,16 @@ static zend_always_inline int zend_verify_property_access(zend_property_info *pr
 	if (property_info->flags & ZEND_ACC_PUBLIC) {
 		return 1;
 	} else if (property_info->flags & ZEND_ACC_PRIVATE) {
-		if (EG(fake_scope)) {
-			scope = EG(fake_scope);
-		} else {
-			scope = zend_get_executed_scope();
+		if (property_info->ce == ce) {
+			if (EG(fake_scope)) {
+				scope = EG(fake_scope);
+			} else {
+				scope = zend_get_executed_scope();
+			}
```

Comment:
This pattern is now used more than 10 times in this file, probably makes sense to create a function to get the (possibly fake) scope.

Reply:
for now, I prefer to keep this, because some EG(fake_scope) accesses may be eliminated